### DISCUSSION TITLE
Problem powering keyboards from TSynth USB host port

### DISCUSSION THREAD

**Discussion by woll** [ElectroTechnique/electrotechnique.github.io#14] [Hardware Issues]
I have 2 (old) M-Audio keyboards and neither of them works if I try to power them through the TSynth USB host port. Should that be possible?

They seem to work OK if I connect them to a USB port on my laptop - they send midi info correctly to apps running on the laptop, and they work OK with TSynth if I power them using their separate 12v power supply.

If I try to power them from the TSynth USB host port (with the TSynth powered from a 1A USB 'charger' or from a laptop), the TSynth hangs, either at the splash 'TSynth" screen, or sometimes a bit further on in boot process. Very occasionally the TSynth finishes the boot process and I can play notes, but a bit later the notes start to get distorted (like rapid modulation).

Both keyboards seem to draw around 100mA max (measured using a USB monitor dongle). The voltage from the laptop USB port is 5v. The voltage at the TSynth USB host port is about 4.87v. The total draw (of TSynth + keyboard) measured at the laptop port is around 215mA.

I've tried googling for things like "teensy usb host limits/inrush" etc and couldn't really find anything that gave any useful info.

Any ideas?

**Comment by Electrotechnique** [comment]
In theory 100mA isn't enough current draw to be a problem - the TPD3S014 limits the current out the host port to 500mA. The voltage - 4.87V seems a bit low however. I would suspect the M-Audio keyboards don't like like a voltage that low. My TSynth supplies about 4.83V on the host port and Arturia Minilab doesn't mind it. Looking at the TPD3S014 datasheet (page 8), the typical output voltage seems to be around 4.8V. The USB standard goes down to 4.75V I think.

A possible solution _which I haven't tried_, could be to break the track to the 5V pin on the host header and connect a wire straight from 5V on the USB. I don't think that this would be a problem with the way the host port works, it wouldn't limit the current draw however.

![Screenshot 2021-02-12 114458](https://user-images.githubusercontent.com/50934180/107726627-d7b6c480-6d2c-11eb-834f-8671013e3cfc.jpg)

**Reply by woll** [reply]
Thanks for the suggestion. I tried it , and supplying the USB host port's 5v directly from the USB cable input to the Teensy does fix the problem.

Looking at the Teensy schematic, it seems that it uses a 100uf capacitor on the TPD3S014 output, rather than the 150uf that is suggested in the data sheet. Not sure if that would make a difference. I tried adding various capacitors across the USB Host 5v and Ground pins (which I think should be parallel to the existing 100uf on the TPD3S014 output) but that didn't seem to make any difference. Maybe it is more down to the voltage rather than the required current.

**Reply by Electrotechnique** [reply]
The capacitor just smooths the output. Really the problem is that some USB connected equipment isn't designed to work down to the USB minimum of 4.75V  (4.40V at hub ports!)
